packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -269,34 +234,48 @@ def _get_provider(self, provider_id: Optional[str]) -> BaseProvider:
 
         return self.providers[provider_id]
 
-    @magic_arguments()
-    @argument('model_id',
-                help="""Model to run, specified as a model ID that may be
-                optionally prefixed with the ID of the model provider, delimited
-                by a colon.""")
-    @argument('-f', '--format',
-                choices=["code", "html", "image", "json", "markdown", "math", "md", "text"],
-                nargs="?",
-                default="markdown",
-                help="""IPython display to use when rendering output. [default="markdown"]""")
-    @argument('-r', '--reset',
-                action="store_true",
-                help="""Clears the conversation transcript used when interacting
-                with an OpenAI chat model provider. Does nothing with other
-                providers.""")
-    @argument('prompt',
-                nargs='*',
-                help="""Prompt for code generation. When used as a line magic, it
-                runs to the end of the line. In cell mode, the entire cell is
-                considered the code generation prompt.""")
+    def handle_help(self, _: HelpArgs):
+        with click.Context(cell_magic_parser, info_name="%%ai") as ctx:
+            click.echo(cell_magic_parser.get_help(ctx))
+        click.echo('-' * 78)
```

Comment:
What does this 78-hyphen line mean? I didn't catch that the line above refers to the `%%ai` cell magic and the line below refers to the `%ai` line magic. It might be good to document this function with comments.

Reply:
Oh, that's a separator between the help for `%%ai` and `%ai`. Without it, the help strings blur together. `click.echo()` is essentially just an implementation of `print()` in the Python stdlib. I don't think it warrants its own comment.